Core/Core/SwiftUIViews/ConfirmationAlert.swift

```diff
@@ -0,0 +1,120 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Combine
+import SwiftUI
+
+public struct ConfirmationAlertViewModel {
+    public let title: String
+    public let message: Text
+    public let cancelButtonTitle: String
+    public let confirmButtonTitle: String
+    public let confirmButtonRole: ButtonRole?
+    public let confirmDidTap = PassthroughSubject<Void, Never>()
+
+    public init(title: String,
+                message: String,
+                cancelButtonTitle: String,
+                confirmButtonTitle: String,
+                isDestructive: Bool = false) {
+        self.title = title
+        self.message = Text(message)
+        self.cancelButtonTitle = cancelButtonTitle
+        self.confirmButtonTitle = confirmButtonTitle
+        self.confirmButtonRole = isDestructive ? .destructive : nil
+    }
+}
+
+public extension View {
+    func confirmationAlert(isPresented: Binding<Bool>,
+                           presenting viewModel: ConfirmationAlertViewModel)
+    -> some View {
+        alertConfirmation(isPresented: isPresented, presenting: viewModel)
+    }
+
+    func alertConfirmation(isPresented: Binding<Bool>,
+                           presenting viewModel: ConfirmationAlertViewModel)
+    -> some View {
+        alert(
+            viewModel.title,
+            isPresented: isPresented,
+            actions: {
+                Button(viewModel.cancelButtonTitle,
+                       role: .cancel,
+                       action: {})
```

Comment:
Wouldn't we need to track cancel in some cases?

Reply:
I added it. Might come handy in the future.